### ISSUE TITLE
[Bug-Candidate]:  Failed to convert IR to SSA for contract.

### ISSUE THREAD

**Issue by MrKevinOConnell** [crytic/slither#2148]
### Describe the issue:

I get this error when I run slither .

### Code example to reproduce the issue:

slither .

### Version:

0.9.6

### Relevant log output:

```shell
Failed to convert IR to SSA for ... contract. Please open an issue https://github.com/crytic/slither/issues.
 
Traceback (most recent call last):
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/__main__.py", line 814, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/__main__.py", line 102, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/__main__.py", line 80, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/slither.py", line 135, in __init__
    self._init_parsing_and_analyses(kwargs.get("skip_analyze", False))
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/slither.py", line 155, in _init_parsing_and_analyses
    raise e
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/slither.py", line 151, in _init_parsing_and_analyses
    parser.analyze_contracts()
  File "/Users/../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 541, in analyze_contracts
    self._convert_to_slithir()
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 774, in _convert_to_slithir
    raise e
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 769, in _convert_to_slithir
    contract.convert_expression_to_slithir_ssa()
  File "/Users/....pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/core/declarations/contract.py", line 1502, in convert_expression_to_slithir_ssa
    func.generate_slithir_ssa(all_ssa_state_variables_instances)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/core/declarations/function_contract.py", line 134, in generate_slithir_ssa
    compute_dominance_frontier(self.nodes)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/core/dominators/utils.py", line 98, in compute_dominance_frontier
    assert runner.immediate_dominator
AssertionError
ERROR:root:Error in .
ERROR:root:Traceback (most recent call last):
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/__main__.py", line 814, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/__main__.py", line 102, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/__main__.py", line 80, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/slither.py", line 135, in __init__
    self._init_parsing_and_analyses(kwargs.get("skip_analyze", False))
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/slither.py", line 155, in _init_parsing_and_analyses
    raise e
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/slither.py", line 151, in _init_parsing_and_analyses
    parser.analyze_contracts()
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 541, in analyze_contracts
    self._convert_to_slithir()
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 774, in _convert_to_slithir
    raise e
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 769, in _convert_to_slithir
    contract.convert_expression_to_slithir_ssa()
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/core/declarations/contract.py", line 1502, in convert_expression_to_slithir_ssa
    func.generate_slithir_ssa(all_ssa_state_variables_instances)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/core/declarations/function_contract.py", line 134, in generate_slithir_ssa
    compute_dominance_frontier(self.nodes)
  File "/Users/.../.pyenv/versions/3.8.16/lib/python3.8/site-packages/slither/core/dominators/utils.py", line 98, in compute_dominance_frontier
    assert runner.immediate_dominator
AssertionError
```

**Comment by 0xalpharush** [comment]
Hi, we have a pending fix in https://github.com/crytic/slither/pull/1984! This issue is a duplicate of https://github.com/crytic/slither/issues/1973 but we'll leave open until we've released a new version with the fix

**Comment by MrKevinOConnell** [comment]
> Hi, we have a pending fix in #1984! This issue is a duplicate of #1973 but we'll leave open until we've released a new version with the fix

awesome! Sorry didn't see that 😬 @0xalpharush

**Comment by 0xalpharush** [comment]
Closed by https://github.com/crytic/slither/pull/1984

**Comment by elopez** [comment]
Hi @blockchain1112 ! Can you upgrade to Slither 0.10.0 and see if the issue persists? The fix was included in that release, and judging by your stack trace, you appear to be running an older version of Slither. You can check your current version by running `slither --version`

**Comment by blockchain1112** [comment]
@elopez 
Did you fix the issue with slither 0.10.0?

**Comment by blockchain1112** [comment]
Thanks @elopez  I have fixed it 
Really thanks for your help
